commands/forward.go

```diff
@@ -0,0 +1,89 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/docker/machine/libmachine"
+	"github.com/docker/machine/libmachine/log"
+)
+
+func cmdForward(c CommandLine, api libmachine.API) error {
+	args := c.Args()
+	if len(args) < 1 || len(args) > 2 {
+		c.ShowHelp()
+		return errWrongNumberArguments
+	}
+
+	target, err := targetHost(c, api)
+	if err != nil {
+		return err
+	}
+
+	address := args[0]
+	if len(args) > 1 {
+		address = args[1]
+	}
+
+	hostInfoLoader := &storeHostInfoLoader{api}
+
+	cmd, err := getForwardCmd(target, address, hostInfoLoader)
+	if err != nil {
+		return err
+	}
+
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	return cmd.Run()
+}
+
+func getForwardCmd(target, address string, hostInfoLoader HostInfoLoader) (*exec.Cmd, error) {
+	var cmdPath string
+	var err error
+
+	cmdPath, err = exec.LookPath("ssh")
+	if err != nil {
+		return nil, err
+	}
+
+	h, err := hostInfoLoader.load(target)
+	if err != nil {
+		return nil, err
+	}
+
+	sshArgs := baseSSHArgs
+	if h.GetSSHKeyPath() != "" {
+		sshArgs = append(sshArgs, "-o", "IdentitiesOnly=yes")
+	}
+
+	port, err := h.GetSSHPort()
+	if err == nil && port > 0 {
+		sshArgs = append(sshArgs, "-p", fmt.Sprintf("%d", port))
+	}
+
+	if h.GetSSHKeyPath() != "" {
+		sshArgs = append(sshArgs, "-o", fmt.Sprintf("IdentityFile=%q", h.GetSSHKeyPath()))
```

Comment:
I'm concerned we'll see the same issue here as the one we had to fix https://github.com/docker/machine/pull/4388 on Windows.

Reply:
Are you saying that `%q` is broken on Windows ? That's unfortunate, but maybe there are workarounds...

Reply:
But you are probably right, since most was just copied from `scp`. Guess I will do the same then.